File: Back to Basics Approach 3rd Edition/CodeStepByStep_java/recursion/countToBy.java
package CodeStepByStep.java.recursion;

/**
 * https://www.codestepbystep.com/problem/view/java/recursion/countToBy
 *
 * @author Alex Golub
 * @since 12-May-21, 6:50 PM
 */
class countToBy {
  public static void main(String[] args) {
    countToBy(10, 1);
    System.out.println();
    countToBy(3, 6);
  }
  
  public static void countToBy(int n, int m) {
    if (n < 1 || m < 1)
      throw new IllegalArgumentException();
    helper(n - m, m);
    System.out.println(n);
  }
  
  private static void helper(int n, int m) {
    if (n >= 1) {
      helper(n - m, m);
      System.out.print(n + ", ");
    }
  }
}

File: Back to Basics Approach 3rd Edition/CodeStepByStep_java/javascript/recursion/countToBy.js
/**
 * https://www.codestepbystep.com/problem/view/javascript/recursion/countToBy
 *
 * @author Alex Golub
 * @since 19-May-21, 9:15 PM
 */
function countToBy(n, m) {
  if (n < 1 || m < 1)
    return '';
  return `${helper(n - m, m)}${n}`;
}

function helper(n, m) {
  if (n >= 1) {
    return helper(n - m, m) + '' + n + ', ';
  } else {
    return '';
  }
}

console.log(countToBy(10, 1));
console.log(countToBy(25, 4));
